02-oop-inheritance/ex5.js

```diff
@@ -0,0 +1,70 @@
+class EventEmitter {
+    constructor() {
+      // listeners object will store events as keys and arrays of listeners as values
+      this.listeners = {};
+    }
+    on(event, listener) {
+        if (this.listeners[event] === undefined) {
+            this.listeners[event] = [];
+        }
+        this.listeners[event].push(listener);
+    }
+    emit(event) {
+        this.listeners[event].forEach(function(fn){
+            console.log("executing a listener");
+            fn.call();
+        })
+    }
+    off(event, listener) {
+        if (this.listeners[event] === undefined){
+            console.log("nothing to remove");
+        }
+        else {
+            let deletedListener = this.listeners[event].pop();
```

Comment:
Also you can avoid `deletedListener` single use variable, just doing:

`console.log("deleted", this.listeners[event].pop());`